src/js/script.js

```diff
@@ -0,0 +1,270 @@
+/*============================================================================================================= */
+
+var slideIndex = 1;
+if (window.location.pathname == "/") {
+    showSlides(slideIndex);
+}
+
+function plusSlides(n) {
+    showSlides(slideIndex += n);
+}
+
+function currentSlide(n) {
+    showSlides(slideIndex = n);
+}
+
+function showSlides(n) {
+    var i;
+    var slides = document.getElementsByClassName("mySlides");
+    var dots = document.getElementsByClassName("dot");
+    if (n > slides.length) {
+        slideIndex = 1;
+    }
+    if (n < 1) {
+        slideIndex = slides.length;
+    }
+    for (i = 0; i < slides.length; i++) {
+        slides[i].style.display = "none";
+    }
+    for (i = 0; i < dots.length; i++) {
+        dots[i].className = dots[i].className.replace(" active", "");
+    }
+    slides[slideIndex - 1].style.display = "block";
+    dots[slideIndex - 1].className += " active";
+}
+
+var carousal = document.getElementById("next");
+setInterval(function() {
+    if (carousal) {
+        carousal.click();
+    }
+}, 5000);
+
+/*================================================================================================ */
+
+/*============================================================================================================== */
+
+// JS for Mobile Category Dropdown
+var ul = null;
+var toggleBtn = '';
+var targ = null;
+var menu = null;
+var list = null;
+window.onload = function() {
+    toggleBtn = document.querySelector('.category-mobile');
```

Comment:
The class names are too generic. And why are you finding everything in document. Its advisable to capture the parent component in a variable. And then use that node to search within that container.

Everytime searching the whole document is a performance hit